Lib/Metaiterators.hpp

```diff
@@ -476,11 +481,49 @@ class MappingIterator
    * the @b size() function, and if the @b knowsSize() function returns true.
    */
   inline size_t size() const { return _inner.size(); }
+
+  auto reverse() && 
```

Comment:
Struggling to parse this. Equivalent to `auto &&reverse() {...}`?

Reply:
No, it is a function that expects that this is a move reference 
Think of
```
auto reverse() && {}
auto reverse() const {}
auto reverse() {}
```
It's best practice to use this in a function that moves out of *this/invalidates it, which means that the user needs to explicitly `std::move` and there is no hidden moving data out of somewhere.